src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py

```diff
@@ -8482,6 +8482,38 @@ def update_mc_profile_default(self) -> ManagedCluster:
         mc = self.update_node_resource_group_profile(mc)
         # update bootstrap profile
         mc = self.update_bootstrap_profile(mc)
+        # update kubernetes version and orchestrator version
+        mc = self.update_kubernetes_version_and_orchestrator_version(mc)
+        return mc
+
+    def update_kubernetes_version_and_orchestrator_version(self, mc: ManagedCluster) -> ManagedCluster:
+        """Update kubernetes version and orchestrator version for the ManagedCluster object.
+
+        :param mc: The ManagedCluster object to be updated.
+        :return: The updated ManagedCluster object.
+        """
+        self._ensure_mc(mc)
+
+        # Check if auto_upgrade_channel is set to "none"
+        auto_upgrade_channel = self.context.get_auto_upgrade_channel()
+        if auto_upgrade_channel == "none":
+            warning_message = (
+                "Since auto-upgrade-channel is set to none, cluster kubernetesVersion will be set to the value of "
+                "currentKubernetesVersion, all agent pools orchestratorVersion will be set to the value of "
+                "currentOrchestratorVersion respectively. Continue?"
+            )
+            if not self.context.get_yes() and not prompt_y_n(warning_message, default="n"):
+                raise CLIError("Operation cancelled by user.")
+
+            # Set kubernetes version to match the current kubernetes version if it has a value
+            if mc.current_kubernetes_version:
+                mc.kubernetes_version = mc.current_kubernetes_version
```

Comment:
Just curious, (like in other clients,) if the user sets the auto upgrade channel to "none" but does not change the kubernetes version accordingly, will the server do this for the user? Or will it return a bad request, or will it just keep the status quo?

Reply:
orchsver validator will deny the request and return operationNotAllowed. 
https://msazure.visualstudio.com/CloudNativeCompute/_git/aks-rp/pullrequest/11940214

Reply:
This cli change is to unblock the cli users, after the the above orchsver validator bug fix. If their cluster are currently on this state (specifiedVersion < currentVersion), If they are trying to disable autoupgrade with cli, they will always fail. 
We are reseting their specified version in cli to pass the validation.